### PR TITLE
add deep rendering tests to workshop form js

### PR DESCRIPTION
**Pull request by davidsbailey**
Related to https://codedotorg.atlassian.net/browse/ACQ-462. This adds a first enzyme `mount` test for the workshop form. For more background, see [workshop form refactor proposal](https://docs.google.com/document/d/1lFUGI_tK7HaTl8VU-WNMjRpIM4kLmThYWkiCYVeg_yg/edit#heading=h.x20ojf9pks8i).

links
* js factory docs: https://www.npmjs.com/package/rosie 

## Testing story

Improves existing test coverage

### CODE REVIEW

**Review by TurnerRiley — APPROVED** [review]
Thank you for the super detailed work through the proposal doc and through setting this up! For my own understanding, what does the <MemoryRouter> component allow us to do here? I'm reading about it now but don't quite understand what it would be doing here?

**Comment by davidsbailey** [comment]
> Thank you for the super detailed work through the proposal doc and through setting this up! For my own understanding, what does the component allow us to do here? I'm reading about it now but don't quite understand what it would be doing here?

Hey Turner, are you asking about the `WorkshopForm` component? this component allows viewing, creating and editing of workshops in the workshop dashboard ui. the screenshots in the doc are a bit small so here they are in a more convenient format. I hope this helps, but please let me know if I misunderstood your question!

### create

Screenshot 2023-04-05 at 12 56 22 PM

### edit 

Screenshot 2023-04-05 at 12 56 42 PM

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
+import {MemoryRouter} from 'react-router-dom';
+import mapboxReducer from '@cdo/apps/redux/mapbox';
+import {createStore, combineReducers} from 'redux';
 
 describe('WorkshopForm test', () => {
-  let fakeWorkshop;
+  let fakeWorkshop, store;
 
   beforeEach(() => {
     fakeWorkshop = Factory.build('workshop');
```
Comment (megcrenshaw) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js:20]: Wow, I didn't know this was a thing!

File: apps/test/unit/code-studio/pd/workshop_dashboard/workshopFactory.js [hunk 2]
```diff
@@ -24,6 +24,15 @@ Factory.define('workshop')
...
   .attr('virtual', false);
 
+Factory.define('csp summer workshop')
+  .extend('workshop')
+  .attr('course', 'CS Principles')
+  .attr('subject', '5-day Summer')
+  .attr('sessions', () => Factory.buildList('session', 5))
+  .attr('account_required_for_attendance?', true)
+  .attr('scholarship_workshop?', true)
+  .attr('location_name', 'physical');
```
Comment (megcrenshaw) [inline apps/test/unit/code-studio/pd/workshop_dashboard/workshopFactory.js:34]: TIL

**Review by megcrenshaw — APPROVED** [review]

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js [hunk 3]
```diff
@@ -1,29 +1,63 @@
...
+      combineReducers({
+        mapbox: mapboxReducer
+      })
+    );
+  });
+
+  it('renders csf intro workshop', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter>
```
Comment (megcrenshaw) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js:31]: Also something new I learned! Thanks for getting this working.

Comment (davidsbailey) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js:31]: I didn't know about it either! GPT-4 discovered it for me:

Screenshot 2023-04-11 at 9 47 27 AM

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js (hunk 1 above)
Comment (davidsbailey) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js:20]: this was also new to me!

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js (hunk 3 above)
Comment (megcrenshaw) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js:31]: Okay wow. That is seriously impressive ... excuse me while I let my head explode and then gather the pieces back together 🤯

**Comment by TurnerRiley** [comment]
> > Thank you for the super detailed work through the proposal doc and through setting this up! For my own understanding, what does the component allow us to do here? I'm reading about it now but don't quite understand what it would be doing here?
> 
> Hey Turner, are you asking about the `WorkshopForm` component? this component allows viewing, creating and editing of workshops in the workshop dashboard ui. the screenshots in the doc are a bit small so here they are in a more convenient format. I hope this helps, but please let me know if I misunderstood your question!
> 
> ### create
> Screenshot 2023-04-05 at 12 56 22 PM
> 
> ### edit
> Screenshot 2023-04-05 at 12 56 42 PM

Oh shoot, I'm not sure what happened by I had typed "the MemoryRouter component" instead of just "the component" but I guess in rephrasing I accidentally deleted that. But you talked more about it with your ChatGPT response to Meg so I understand now, thank you!

**Comment by davidsbailey** [comment]
> Oh shoot, I'm not sure what happened by I had typed "the MemoryRouter component" instead of just "the component" but I guess in rephrasing I accidentally deleted that. But you talked more about it with your ChatGPT response to Meg so I understand now, thank you!

ahh, that makes more sense! yeah, it appears to be designed for testing with react router: https://reactrouter.com/en/main/router-components/memory-router

**Pull request merged**